install.sh

```diff
@@ -194,9 +190,41 @@ __install_bro() {
 	__install_packages bro broctl
 	chmod 2755 /opt/bro/logs
 	_BRO_PKG_INSTALLED=true
+	_BRO_INSTALLED=true
 	_BRO_PATH="/opt/bro/bin"
 }
 
+__configure_bro() {
+	# Configure Bro
+	curl -sSL "https://raw.githubusercontent.com/activecm/bro-install/master/gen-node-cfg.sh" -o "./gen-node-cfg"
+	curl -sSL "https://raw.githubusercontent.com/activecm/bro-install/master/node.cfg-template" -o "./node.cfg-template"
+	chmod 755 ./gen-node-cfg
+	_BRO_CONFIGURED=true
+	./gen-node-cfg || _BRO_CONFIGURED=false
+	rm -f "./gen-node-cfg"
+	rm -f "./node.cfg-template"
+
+	if [ "$_BRO_CONFIGURED" = "true" ]; then
+		printf "\n$_IMPORTANT Enabling Bro on startup.\n"
+		# don't add a new line if broctl is already in cron
+		if [ $(crontab -l 2>/dev/null | grep 'broctl cron' | wc -l) -eq 0 ]; then 
+			# Append an entry to an existing crontab
+			# crontab -l will print to stderr and exit with code 1 if no crontab exists
+			# Ignore stderr and force a successfull exit code to prevent an install error			
+			(crontab -l 2>/dev/null || true; echo "*/5 * * * * $_BRO_PATH/broctl cron") | crontab
+		fi
+		$_BRO_PATH/broctl cron enable > /dev/null
+		printf "$_IMPORTANT Enabling Bro on startup process completed.\n"
+
+		printf "$_IMPORTANT Starting Bro. \n"
+		$_BRO_PATH/broctl deploy
```

Comment:
If a security onion user hasn't run the setup program, the installer will crash as broctl is broken until setup is ran.

Reply:
The gen-node-cfg script prompts the user with the proposed changes and asks if they want to overwrite. And it saves the previous node.cfg as a dated backup in the same directory. That should cover the rare cases right? A user could just restore from the backup if they accidentally applied an incorrect config?

Reply:
Any suggestions on how to detect the broken broctl?  Does the binary exist?  Can we run `-v` or something and check for error output first?

Reply:
I don't know if there is an easy way to detect if the security onion setup command has been run or not.

Reply:
The first case I described would not be fixed by the overwrite prompt. The first case happens because we call `broctl deploy` without asking the user. The deploy function takes the current broctl config and sets up bro from that config. If the user had a good broctl config, deployed, broke their broctl config, and went on about their days, they would lose their working bro install when we call `broctl deploy`.

Reply:
The broken broctl happens in security onion because the setup program has not ran yet. Broctl is set to run under the user `sguil`, but the user `sguil` has not been created yet. `sguil` is a system account for the Sguil analyst console program that is installed as a part of the security-onion setup script.